backend_new/src/services/application.service.ts

```diff
@@ -195,68 +211,328 @@ export class ApplicationService {
+              create: {
+                ...dto.accessibility,
+              },
+            }
+          : undefined,
+        alternateContact: dto.alternateContact
+          ? {
+              create: {
+                ...dto.alternateContact,
+                address: {
+                  create: {
+                    ...dto.alternateContact.address,
+                  },
+                },
+              },
+            }
+          : undefined,
+        applicationsAlternateAddress: dto.applicationsAlternateAddress
+          ? {
+              create: {
+                ...dto.applicationsAlternateAddress,
+              },
+            }
+          : undefined,
+        applicationsMailingAddress: dto.applicationsMailingAddress
+          ? {
+              create: {
+                ...dto.applicationsMailingAddress,
+              },
+            }
+          : undefined,
+        listings: dto.listings
+          ? {
+              connect: {
+                id: dto.listings.id,
+              },
+            }
+          : undefined,
+        demographics: dto.demographics
+          ? {
+              create: {
+                ...dto.demographics,
+              },
+            }
+          : undefined,
+        preferredUnitTypes: dto.preferredUnitTypes
+          ? {
+              connect: dto.preferredUnitTypes.map((unitType) => ({
+                id: unitType.id,
+              })),
+            }
+          : undefined,
+        householdMember: dto.householdMember
+          ? {
+              create: dto.householdMember.map((member) => ({
+                ...member,
+                sameAddress: member.sameAddress || YesNoEnum.no,
+                workInRegion: member.workInRegion || YesNoEnum.no,
+                householdMemberAddress: {
+                  create: {
+                    ...member.householdMemberAddress,
+                  },
+                },
+                householdMemberWorkAddress: {
+                  create: {
+                    ...member.householdMemberWorkAddress,
+                  },
+                },
+              })),
+            }
+          : undefined,
+        programs: JSON.stringify(dto.programs),
+        preferences: JSON.stringify(dto.preferences),
       },
+      include: view.details,
     });
 
-    if (!rawApplication) {
-      throw new NotFoundException(
-        `applicationId ${applicationId} was requested but not found`,
-      );
+    if (dto.applicant.emailAddress && forPublic) {
+      // TODO: email service https://github.com/bloom-housing/bloom/pull/3607
     }
 
     return mapTo(Application, rawApplication);
   }
 
   /*
-    this will create an application
+    this will update an application
+    if no application has the id of the incoming argument an error is thrown
   */
-  async create(incomingData: any) {
-    // TODO
+  async update(dto: ApplicationUpdate): Promise<Application> {
+    const rawApplication = await this.findOrThrow(dto.id);
```

Comment:
`rawApplication` isn't used. Is the intent to just check for the existence of it?

Reply:
This will get used during the perm checking that happens just below this

resolving convo!